app/controllers/images_controller.rb

```diff
@@ -72,28 +72,44 @@ def show
 
   def update
     @warpable = Warpable.find params[:warpable_id]
-
-    nodes = []
-    author = @warpable.map.author
-
-    # is it really necessary to make new points each time?
-    params[:points].split(':').each do |point|
-      lon = point.split(',')[0]
-      lat = point.split(',')[1]
-      node = Node.new(color: 'black',
-                      lat: lat,
-                      lon: lon,
-                      author: author,
-                      name: '')
-      node.save
-      nodes << node
+    map = Map.find(@warpable.map_id)
+    if map.anonymous? || logged_in?
+      nodes = []
+      author = @warpable.map.author
+      # is it really necessary to make new points each time?
+      params[:points].split(':').each do |point|
+        lon = point.split(',')[0]
+        lat = point.split(',')[1]
+        node = Node.new(color: 'black',
+                        lat: lat,
+                        lon: lon,
+                        author: author,
+                        name: '')
+        node.save
+        nodes << node
+      end
+      @warpable.count_version = 0
+      @warpable.nodes = nodes.collect(&:id).join(',')
+      @warpable.locked = params[:locked]
+      @warpable.cm_per_pixel = @warpable.get_cm_per_pixel
+      @warpable.save
+      render html: 'success'
+    else
+      render plain: 'You must be logged in to update the image, unless the map is anonymous.'
     end
+  end
 
-    @warpable.nodes = nodes.collect(&:id).join(',')
-    @warpable.locked = params[:locked]
-    @warpable.cm_per_pixel = @warpable.get_cm_per_pixel
+  def revert
+    @warpable = Warpable.find params[:id]
+    version = @warpable.versions.find(params[:version])
+    if(@warpable.count_version==1)
+      version.reify&.save
+    else
```

Comment:
Hey @divyabaid16, just read @jywarren's comments. If we want to save cloned versions then if condition is not required

Reply:
I guess,  count_version has no role to play now.

Reply:
Also, are you available for a quick call?

Reply:
Ya, Count_version has no role to play now. I will remove it.

Reply:
ok, so let's join on hangout?

Reply:
Okay @gauravano .